src/com/griddynamics/cinema/Cinema.java

```diff
@@ -0,0 +1,82 @@
+package com.griddynamics.cinema;
+
+import java.util.stream.Stream;
+
+public class Cinema {
+	
+    final char[][] HALL;
+    final int ROWS;
+    final int SEATS_IN_EACH_ROW;
+    
+    private final int SEATS_COUNT;
+    private final int TOTAL_INCOME;
+
+    private int ticketsPurchased = 0;
+    private int currentIncome = 0;
+    
+    
+    
+    Cinema(int rows, int seatsInEachRow) {
+    	
+    	ROWS = rows;
+    	SEATS_IN_EACH_ROW = seatsInEachRow;
+
+        SEATS_COUNT = ROWS * SEATS_IN_EACH_ROW;
+        TOTAL_INCOME = SEATS_COUNT * 10
+                - (SEATS_COUNT < 60 ? 0 : (rows + 1) / 2 * seatsInEachRow * 2);
+
+        HALL = Stream.generate(() -> "S".repeat(SEATS_IN_EACH_ROW).toCharArray())
+                .limit(ROWS).toArray(x -> new char[ROWS][]);
+    }
+    
+    
+    
+    private void increaseSoldTicketsAmount() {
+    	++ticketsPurchased;
+    }
+    private void increaseCurrentIncome(int ticketPrice) {
+    	currentIncome += ticketPrice;
+    }
+    private void reserveTheSeat(int row, int seat){
+    	HALL[row][seat] = 'B';
+    }
+    
+    
+    
+    void showSeats() {
+
+        System.out.println("\nCinema:\n"
+                + "  1 2 3 4 5 6 7 8 9".substring(0, SEATS_IN_EACH_ROW * 2 + 1));
```

Comment:
what happens if seats in each row will be 10?

Reply:
In description of one of projects stages was declared condition that number of rows and number of seats would be less than 10.

Reply:
I think it is not the best way how to generate string of numbers, how you can change it?

Reply:
Can I use this?
System.out.print(IntStream.generate(new AtomicInteger()::getAndIncrement)
				.limit(100/**/).boxed().map(String::valueOf).collect(Collectors.joining(" ")));

Reply:
Can I use this:
System.out.print(
IntStream.generate(new AtomicInteger()::getAndIncrement)
.limit(/*random amount of rows*/)
.boxed()
.map(String::valueOf)
.collect(Collectors.joining(" "))
);
?

Reply:
Can I use this:

		System.out.print(
				IntStream.generate(new AtomicInteger()::getAndIncrement)
				.limit(/*random amount of rows*/)
				.boxed()
				.map(String::valueOf)
				.collect(Collectors.joining(" ")));

?

Reply:
Can I use this:

	System.out.print(
			IntStream.generate(new AtomicInteger()::getAndIncrement)
			.limit(/*random amount of seats*/)
			.boxed()
			.map(String::valueOf)
			.collect(Collectors.joining(" ")));
?

Reply:
By using Stream API, I think. For example:

		 System.out.println(Stream.iterate(1, i -> i + 1)
				 .limit(/*random amount of seats*/)
				 .map(String::vlaueOf)
				 .collect(Collectors.joining(" ", "   ", "")));

Reply:
By using Stream API, I think. For example:

    	System.out.println("\nCinema\n"
    		+ Stream.iterate(1, i -> i + 1)
    			.limit(/*random amount of seats*/1)
    			.map(String::valueOf)
    			.collect(Collectors.joining(" ", "   ", "")));

Reply:
By using Stream API, I think. For example:

	System.out.println("\nCinema\n"
		+ Stream.iterate(1, i -> i + 1)
			.limit(/*random amount of seats*/)
			.map(String::valueOf)
			.collect(Collectors.joining(" ", "   ", "")));